pkg/server/registry/apigroups/acorn/imageallowrules/strategy.go

```diff
@@ -0,0 +1,88 @@
+package imageallowrules
+
+import (
+	"context"
+
+	apiv1 "github.com/acorn-io/acorn/pkg/apis/api.acorn.io/v1"
+	internalv1 "github.com/acorn-io/acorn/pkg/apis/internal.acorn.io/v1"
+
+	"github.com/acorn-io/mink/pkg/types"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apiserver/pkg/storage"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Strategy struct {
+	client client.WithWatch
+}
+
+func NewStrategy(c client.WithWatch) *Strategy {
+	return &Strategy{client: c}
+}
+
+func (s *Strategy) NewList() types.ObjectList {
+	return &apiv1.ImageAllowRulesList{}
+}
+
+func (s *Strategy) New() types.Object {
+	return &apiv1.ImageAllowRules{}
+}
+
+func (s *Strategy) List(ctx context.Context, namespace string, options storage.ListOptions) (types.ObjectList, error) {
+	clusterImageAllowRules := &internalv1.ClusterImageAllowRulesInstanceList{}
+	if err := s.client.List(ctx, clusterImageAllowRules); err != nil {
+		return nil, err
+	}
+
+	projectImageAllowRules := &internalv1.ProjectImageAllowRulesInstanceList{}
+	if err := s.client.List(ctx, projectImageAllowRules, &client.ListOptions{Namespace: namespace}); err != nil {
+		return nil, err
+	}
+
+	imageAllowRules := apiv1.ImageAllowRulesList{Items: make(
+		[]apiv1.ImageAllowRules,
+		0,
+		len(clusterImageAllowRules.Items)+len(projectImageAllowRules.Items))}
+
+	for _, cic := range clusterImageAllowRules.Items {
+		imageAllowRules.Items = append(imageAllowRules.Items, apiv1.ImageAllowRules{
+			ObjectMeta: v1.ObjectMeta{Name: cic.Name, CreationTimestamp: cic.CreationTimestamp},
```

Comment:
How are cluster and project objects with the same name handled? Does the project object take precedence? Would it be possible for a user to select a cluster image allow rule when a project image allow rule with the same name exists?

All of these questions lead to: should we not list a cluster allow rule if a project allow rule with the same name exists?

Reply:
They're additive, so you cannot "replace" or negate a cluster rule with a project role, you can only tighten the rules, not loosen them.

> Would it be possible for a user to select a cluster image allow rule when a project image allow rule with the same name exists?

What do you mean by "selecting a rule" here?

Reply:
What I am asking here is about cluster rules and project rules with the same name. It seems that if there is a cluster rule and project rule with the same name, then all of them are being returned here. So, for example, if a user lists all the image allow rules, they may see duplicated names/configurations.

There was a similar issue with compute/volume classes. We made the choice that if a project object has the same name as a cluster object, then the project object would be returned/used and not the cluster object.

Should the same rule apply here?

Reply:
As per latest alignment, we will only continue with project-level resources for now.